src/dbnode/network/server/tchannelthrift/node/service_test.go

```diff
@@ -394,6 +450,41 @@ func TestServiceFetchIsOverloaded(t *testing.T) {
 	require.Equal(t, tterrors.NewInternalError(errServerIsOverloaded), err)
 }
 
+func TestServiceFetchUnknownNs(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockDB := storage.NewMockDatabase(ctrl)
+	mockDB.EXPECT().Options().Return(testStorageOpts).AnyTimes()
+	mockDB.EXPECT().IsOverloaded().Return(false)
+
+	service := NewService(mockDB, testTChannelThriftOptions).(*service)
+	tctx, _ := tchannelthrift.NewContext(time.Minute)
+	ctx := tchannelthrift.Context(tctx)
+	defer ctx.Close()
+
+	start := time.Now().Add(-2 * time.Hour)
+	end := start.Add(2 * time.Hour)
+
+	start, end = start.Truncate(time.Second), end.Truncate(time.Second)
+	nsID := "metrics"
+	unknownErr := fmt.Errorf("unknown-err")
+
+	mockDB.EXPECT().
+		ReadEncoded(ctx, ident.NewIDMatcher(nsID), ident.NewIDMatcher("foo"), start, end).
+		Return(nil, unknownErr)
+
+	_, err := service.Fetch(tctx, &rpc.FetchRequest{
```

Comment:
I would expect this test to return `unknown namespace: metrics` since it's called `TestServiceFetchUnknownNs`. Same for above (`TestServiceQueryUnknownNs`).

Reply:
renamed the tests to be called `...UnknownErr`. Intentionally using a different type of error in these tests, this way we have fewer dependencies across packages.